sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/dsl/package.scala

```diff
@@ -162,6 +162,8 @@ package object dsl {
     def stddev(e: Expression): Expression = Stddev(e)
     def stddev_pop(e: Expression): Expression = StddevPop(e)
     def stddev_samp(e: Expression): Expression = StddevSamp(e)
+    def collect_list(e: Expression): Expression = CollectList(e)
+    def collect_set(e: Expression): Expression = CollectSet(e)
```

Comment:
This is really only for easy building of test cases.  To make it user facing, just add it to [fuctions.py](https://github.com/apache/spark/blob/master/python/pyspark/sql/functions.py) and [functions.scala](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/functions.scala)